### ISSUE TITLE
Problem creating PDF when queried field is empty in database

### ISSUE THREAD

**Issue by KwilliamAB** [jorisschellekens/borb#61]
Hi, any suggestions on how to fix the below error, when a database query has a blank reference cell (without using null=True or blank=False)?  In a Django app.

When pulling data from sqlite3 db, and the referenced field is blank, I am getting the following error:

Exception Type: AssertionError 
Exception Value: A Rectangle must have a non-negative height.

My borb code snippet:

    `layout.add(FixedColumnWidthTable(number_of_columns=4, number_of_rows=4)
        .add(TableCell(Paragraph("Education and Professional Designations", font='Helvetica-bold',
                        text_alignment=Alignment.CENTERED),
                        background_color=table_header_color,
                        col_span=4))

        .add(TableCell(Paragraph("Education:", font_size=11), col_span=1, row_span=3))
        .add(TableCell(Paragraph(education_type_1, font_size=10), col_span=1))
        .add(TableCell(Paragraph(education_notes_1, font_size=10), col_span=2)))`   # <--- this line is causing the issue when cell is empty in database

From Django model:
    education_type_1 = models.CharField(max_length=30, choices=education_choices, blank=True)
    education_notes_1 = models.CharField(max_length=30, blank=True) 

However, if I set the Paragraph string to " ", the PDF creates as expected:

 `layout.add(FixedColumnWidthTable(number_of_columns=4, number_of_rows=4)
        .add(TableCell(Paragraph("Education and Professional Designations", font='Helvetica-bold',
                        text_alignment=Alignment.CENTERED),
                        background_color=table_header_color,
                        col_span=4))

        .add(TableCell(Paragraph("Education:", font_size=11), col_span=1, row_span=3))
        .add(TableCell(Paragraph(education_type_1, font_size=10), col_span=1))
        .add(TableCell(Paragraph(" ", font_size=10), col_span=2)))`

Any guidance would be much appreciated.

Thanks

**Comment by jorisschellekens** [comment]
Are you passing `None` to the constructor of `Paragraph`?

**Comment by KwilliamAB** [comment]
Hi, I believe just a blank string is being passed to the `Paragraph` constructor.  If I pass none/null, "None" prints out on my exported PDF for that field, which is unwanted.

**Comment by jorisschellekens** [comment]
You're not allowed to pass `None` or `empty str` to a `Paragraph`.
It raises all kinds of questions:

- Should the `Paragraph` have a height? How tall should it be? Is `empty str` as tall as `a` or more like `M`?
- How wide should the resulting `Paragraph` be?
- Either the `Paragraph` has no width/height (in which case you could reasonably ask yourself the question why you are rendering it) or it has some surface area, which means it may even potentially cause line-breaks, table-realignment, etc.

In short: **Check whether you are passing `empty str` to a `Paragraph`. And don't do it.**

**Comment by KwilliamAB** [comment]
Ok yeah, that makes sense.  Thank you for being so responsive.
